### PR TITLE
Add a Java base image.

### PR DESCRIPTION
**Pull request by dlorenc**

### CODE REVIEW

File: java/BUILD [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+docker_build(
+    name = "java8",
+    base = "//base:base",
+    debs = [
+        "@zlib//file",
+        "@libfreetype//file",
+        "@libpng//file",
+        "@openjdk_jre8//file",
+        "@libgcc1//file",
```
Comment (mattmoor) [inline java/BUILD]: Do you want to just split off the CPP base?

File: java/BUILD [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+load("@io_bazel_rules_docker//docker:docker.bzl", "docker_build")
+
+docker_build(
+    name = "java8",
+    base = "//base:base",
+    debs = [
+        "@zlib//file",
+        "@libfreetype//file",
+        "@libpng//file",
```
Comment (mattmoor) [inline java/BUILD]: I think libfreetype and libpng were me messing around trying to get Jenkins running.  I don't think they're needed for a properly minimal setup.

File: WORKSPACE [hunk 3]
```diff
@@ -35,6 +35,37 @@ http_file(
     sha256 = "d3819ceeb2becc96aea4b0fceda48f6e68f84193526bc67f6895c00796ee99d6",
 )
 
+# For Java
+http_file(
+   name = "zlib",
+   url = "http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-2+b1_amd64.deb",
```
Comment (mattmoor) [inline WORKSPACE]: add checksums?

Comment (dlorenc) [inline WORKSPACE]: Done.

File: java/BUILD (hunk 2 above)
Comment (dlorenc) [inline java/BUILD]: Ah nice.

**Pull request merged**